packages/base-styles/_mixins.scss

```diff
@@ -438,50 +438,49 @@
 	}
 }
 
-@mixin admin-scheme($color-primary) {
-	--wp-admin-theme-color: #{$color-primary};
-
-	// Ideally, we should limit these computed variables,
-	// because we can"t compute colors based on CSS variables.
-	// This requires refactoring some existing styles.
-	--wp-admin-theme-color-darker-20: #{darken($color-primary, 20%)};
-	--wp-admin-theme-color-darker-10: #{darken($color-primary, 10%)};
-	--wp-admin-theme-color-lighter-10: #{lighten($color-primary, 10%)};
-	--wp-admin-theme-color-lighter-40: #{lighten($color-primary, 40%)};
+@mixin admin-scheme($color-primary-h, $color-primary-s, $color-primary-l) {
+	--wp-admin-theme-color: hsl(#{$color-primary-h}, #{$color-primary-s}, #{$color-primary-l});
+
+	// Darker shades.
+	--wp-admin-theme-color-darker-10: hsl(#{$color-primary-h}, #{$color-primary-s}, #{$color-primary-l - 5});
+	--wp-admin-theme-color-darker-20: hsl(#{$color-primary-h}, #{$color-primary-s}, #{$color-primary-l - 10});
```

Comment:
I'd love to get rid of these two variables as well at some point to allow changing the colors by just defining a single CSS variable :) 

This is still a great improvement thought. I don't think hsl bring value though since we're able to use SASS here. What's the reasoning?

Reply:
I could make it a mixin.

Hsl is native browser, so we can actually retire the variables and have lighter and darker colors from a single color.

Will try in a bit

Reply:
> Hsl is native browser, so we can actually retire the variables and have lighter and darker colors from a single color.

We could but only if we use three CSS variables instead of one which IMO is not great.

Reply:
I do see your point. 

You could limit it to two:

```
--wp-admin-theme-color-hs: '200, 100%';
--wp-admin-theme-color: hsl( var(--wp-admin-theme-color-hs), 36% );
```

Then you could stitch together variables in the code like this, instead of defining them as new colors:

```
button {
	background-color: hsl( var(--wp-admin-theme-color-hs), 26% ); // 10% darker
}
```

You could even do this:

```
$wp-admin-theme-color-10: hsl( var(--wp-admin-theme-color-hs), 26% );
```

Make sense?

Reply:
I think it's better but If I want to tweak the color, I'd have to define `--wp-admin-theme-color-hs` instead of `--wp-admin-theme-color` right?

I mean, we're kind of forcing third-party API to use HSL too; Maybe that's fine but how confident are we that this will last forever?

Reply:
i guess what I'm saying is that i personally prefer to just leave it as is for now. Avoid hsl entirely.

Reply:
Alright, I don't disagree. I've reverted the HSL magic for now, and we're back to having two additional dark colors. 

We _could_ remove one of them because a hover color isn't strictly necessary. But for the time being, I've kept it.